R/GRanges_helpers.R

```diff
@@ -102,22 +103,47 @@ strandPerGroup <- function(grl, keep.names = T){
   }
 }
 
-#' get first exon per granges group
+#' helper function to get a logical list of True/False, if GRangesList group have
+#' + strand = T, if - strand = F
+#' Also checks for * strands, so a good check for bugs
+#' @param grl a GRangesList or GRanges object
+strandBool <- function(grl){
+  if(class(grl) == "GRanges"){
+    posIndeces <- as.character(strand(grl)) == "+"
+  } else {
+    posIndeces <- strandPerGroup(grl, F) == "+"
+  }
+
+  sums <- sum(posIndeces) + sum(!posIndeces)
+  if(is.na(sums)){
+    stop("could not get strands from grl object,\n
```

Comment:
This is in general not needed. We can assume users are going to pass into arguments whatever the function arguments are specified, if user inputs NULL or some other data type he will get errors but its not a concern for us, he is trying something non standard, he expects things to go wrong. Remove this kind of checks everywhere as they are completely unnecessary, they add code and slow down evaluation.